src/LoginForm.jsx

```diff
@@ -1,27 +1,52 @@
-import React from 'react';
-import { login } from './api';
+import React, { useState } from "react";
+import { login } from "./api";
 
 // ======================== LOGIN FORM ======================
 //  * You have an incomplete login form
 //  * You are not allowed to add any additional HTML element
 //  * You are not allowed to use refs
-// 
-// 
+//
+//
 // Tasks:
-// * The "Login" button should trigger the "login()" action imported above and should pass the required data
-// * Disable the "Login" button if email is blank OR if password is less than 6 letters
-// * Disable the "Login" button while "login()" action is being performed
-// * Show an error message from the "login()" action is being performed. The error message should be cleared every time user re-attempts to login
-// * Show an alert box (native Javascript alert) if login succeeds. CHECK THE "login()" FUNCTION TO FIND OUT HOW TO LOGIN SUCCESSFULLY.
+// * +The "Login" button should trigger the "login()" action imported above and should pass the required data
+// * +Disable the "Login" button if email is blank OR if password is less than 6 letters
+// * +Disable the "Login" button while "login()" action is being performed
+// * +Show an error message from the "login()" action is being performed. The error message should be cleared every time user re-attempts to login
+// * +Show an alert box (native Javascript alert) if login succeeds. CHECK THE "login()" FUNCTION TO FIND OUT HOW TO LOGIN SUCCESSFULLY.
 
 const LoginForm = () => {
+  const [formValue, setFormValue] = useState({
+    email: "test@gmail.com",
+    password: "password",
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const onSubmit = async (e) => {
+    try {
+      e.preventDefault();
+      setIsLoading(true);
+      await login(formValue.email, formValue.password);
+      setIsLoading(false);
```

Comment:
` setIsLoading(false);` is used in both "try" and "catch" blocks. You can define it in `finally` block, so you do not repeat yourself. And overall handling function would be something like this:

````  const handleSubmit = async (event) => {
    event.preventDefault();
    setLoading(true);
    setError('');
    try {
      const response = await login(formData.email, formData.password);
      alert(`Message: ${response.message}`);
    } catch (error) {
      setError(error);
    } finally {
      setLoading(false);
    }
  };
````

Reply:
Please check my latest commit which is fix: isLoading state, Added the state of isLoading